lib/Raven/Serializer.php

```diff
@@ -67,17 +67,44 @@ public function __construct($mb_detect_order = null)
      * sanitization and encoding.
      *
      * @param mixed $value
+     * @param array $context
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, array $context = [])
+    {
+        $full_context = $this->get_full_context($context);
+        return $this->serialize_inner($value, $full_context['max_depth'], 0);
+    }
+
+    /**
+     * @param array $context
+     * @return array
+     */
+    protected function get_full_context(array $context)
+    {
+        if (!array_key_exists('max_depth', $context)) {
```

Comment:
>are still public so they could be invoked

No, we don't. It's obvious that the `Serializer::serialize` method is public because it must be called, but the instance itself of the serializer **should not** be accessed outside of the Client class because it's used internally and it's not meant to be called by 3rd parties. There are now two methods to retrieve the serializer and representation serializer instances because I had no choice while refactoring the `Stacktrace` class, but the plan is to remove them when we will refactor the `Client` class